src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1211,9 +1211,9 @@ public function isSecure()
     public function getHost()
     {
         if ($this->isFromTrustedProxy() && self::$trustedHeaders[self::HEADER_CLIENT_HOST] && $host = $this->headers->get(self::$trustedHeaders[self::HEADER_CLIENT_HOST])) {
-            $elements = explode(',', $host);
+            $elements = explode(',', $host, 2);
 
-            $host = $elements[count($elements) - 1];
+            $host = $elements[0];
```

Comment:
Because the port and scheme info are taken from the other edge, so this looks even more suspicious to me :) if the goal is to give back the user facing host, this is currently the wrong side.
I'd be happy to have eg @stof give his PV here.